packages/devtools_app/lib/src/code_size/code_size_attribution.dart

```diff
@@ -0,0 +1,310 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:vm_snapshot_analysis/precompiler_trace.dart';
+import 'package:vm_snapshot_analysis/program_info.dart';
+
+import '../common_widgets.dart';
+import '../table.dart';
+import '../table_data.dart';
+import '../theme.dart';
+import '../trees.dart';
+import '../utils.dart';
+
+class CallGraphWithDominators extends StatefulWidget {
+  const CallGraphWithDominators({@required this.callGraphRoot});
+
+  final CallGraphNode callGraphRoot;
+
+  @override
+  _CallGraphWithDominatorsState createState() =>
+      _CallGraphWithDominatorsState();
+}
+
+class _CallGraphWithDominatorsState extends State<CallGraphWithDominators> {
+  bool showCallGraph = false;
+
+  DominatorTreeNode dominatorTreeRoot;
+
+  @override
+  void initState() {
+    super.initState();
+    dominatorTreeRoot =
+        DominatorTreeNode.from(widget.callGraphRoot.dominatorRoot);
+  }
+
+  @override
+  void didUpdateWidget(covariant CallGraphWithDominators oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.callGraphRoot != widget.callGraphRoot) {
+      dominatorTreeRoot =
+          DominatorTreeNode.from(widget.callGraphRoot.dominatorRoot);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        areaPaneHeader(
+          context,
+          title: showCallGraph ? 'Call Graph' : 'Dominator Tree',
+          needsTopBorder: false,
+          needsBottomBorder: false,
+          needsLeftBorder: true,
+          actions: [
+            const Text('Show call graph'),
+            Switch(
+              value: showCallGraph,
+              onChanged: _toggleShowCallGraph,
+            ),
+          ],
+        ),
+        Expanded(
+          child: showCallGraph
+              ? CallGraphView(node: widget.callGraphRoot)
+              : DominatorTree(
+                  dominatorTreeRoot: dominatorTreeRoot,
+                  selectedNode: widget.callGraphRoot,
+                ),
+        ),
+      ],
+    );
+  }
+
+  void _toggleShowCallGraph(bool shouldShow) {
+    setState(() {
+      showCallGraph = shouldShow;
+    });
+  }
+}
+
+class CallGraphView extends StatefulWidget {
+  const CallGraphView({@required this.node});
+
+  static const Key fromTableKey = Key('CallGraphView - From table');
+  static const Key toTableKey = Key('CallGraphView - To table');
+
+  final CallGraphNode node;
+
+  @override
+  _CallGraphViewState createState() => _CallGraphViewState();
+}
+
+class _CallGraphViewState extends State<CallGraphView> {
+  final _fromColumn = FromColumn();
+
+  final _toColumn = ToColumn();
+
+  CallGraphNode node;
```

Comment:
make this a getter 
e.g. 
```dart
CllGraphNode get node => widget.node;
```
That way you can delete the entire didUpdateWidget method.

Reply:
nevermind. saw where this is modified. I would change `widget.node` to `root` or change this to `selectedNode`.

Reply:
renamed